meter/tibber-pulse.go

```diff
@@ -141,32 +158,69 @@ func NewTibberFromConfig(ctx context.Context, other map[string]interface{}) (api
 
 	go func() {
 		<-ctx.Done()
+		log.INFO.Println("context canceled, closing Tibber Pulse client")
 		if err := client.Close(); err != nil {
-			log.ERROR.Println(err)
+			log.ERROR.Printf("error closing Tibber Pulse client: %v", err)
 		}
 	}()
 
+	reconnectCount := 0
 	go func() {
 		// The pulse sometimes declines valid(!) subscription requests, and asks the client to disconnect.
 		// Therefore we need to restart the client when exiting gracefully upon server request
 		// https://github.com/evcc-io/evcc/issues/17925#issuecomment-2621458890
-		for tick := time.Tick(10 * time.Second); ; {
-			if err := client.Run(); err != nil {
-				log.ERROR.Println(err)
+
+		// Note that there are several reconnection strategies in play:
+		// 1. Mechanism built into Hasura go graphql client
+		// 2. This loop, which is triggered server when Hasura exits on error or gracefully
+		// 3. evcc itself restarts if the client exits with an error
+
+		// Exponential backoff parameters
+		baseDelay := 5 * time.Second
```

Comment:
Going forward we should use centalki/backoff/v4 but let's concentrate on the output first.